chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/DeveloperManagerImpl.java

```diff
@@ -221,6 +225,17 @@ public DeveloperDTO update(final DeveloperDTO updatedDev, final boolean doValida
         }
 
         DeveloperDTO beforeDev = getById(updatedDev.getId());
+
+        // ROLE_ACB cannot edit Transparency Attestation as per OCD-3164
+        if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE_PLUS_ONE_WEEK)) {
```

Comment:
A lot of our validation stuff has been foisted off on validation classes, and removed from the managers themselves. I don't know if this needs to be placed there too, but wanted to mention it

I think the idea is to try to move validation / security out of managers into reviewers, but, TBH, a lot of the lines are blurring, so I don't know for sure. @kekey1 @tmy1313 , thoughts?

Reply:
Interesting. I hadn't considered this a validation but instead just authentication and I wanted the authentication to be pre-validated so it was working with valid info. I consulted the current flow of the existing code which follows this pattern. Scrolling down in the same method shows various snippets which generally do similar things in the same method (check for authentication of some role and/or if values have changed and throw EntityCreationExceptions),. So I'm thiking if we wanted to move my new code to validation, that's fine, but we'd probably want to move the rest too.

Reply:
I'd be fine with moving this into a permissions class and refactoring other code in the same method to be in there as well but would like to be conscious of not letting it get to be too much for this one ticket. Consider how much regression testing would have to happen if we move the code (I don't know how much it is to be moved) and update test cases accordingly if we decide to go this route.

Reply:
I see now you said slash security. If there is some externalized security area maybe we move it and the other code there.

Reply:
In response to Katy, I agree on the possible issue with ticket size i this case because it is a 2 part ticket and has a front-end side as well that must be done after the backend is tested. That being said, I don't personally object, it will just take time. And I don't know that there is any basis for  a security/auth/permissions class yet or if it would be something brand new? Sounds new?